File: Python/Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        
        map ={}

        for i in range(len(nums)):
            current = nums[i]
            complement = target - current
            if complement in map:
                return [map[complement],i]
                
            map[current] = i

File: Javascript/Two Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {

    const map = {};

    for(let i = 0; i < nums.length; i++){
        let current = nums[i];
        let complement = target - current;
        if(map[complement] !== undefined){
            return [map[complement], i];
        }

        map[current] = i;
    }
}
